src/workerd/server/server.c++

```diff
@@ -674,7 +674,7 @@ public:
 
 private:
   kj::Own<kj::Network> network;
-  kj::Maybe<kj::Own<kj::Network>> tlsNetwork;
+  kj::Maybe<kj::Own<kj::SecureNetworkWrapper>> tlsNetwork;
```

Comment:
This is the breaking API change I was talking about in the capnp PR review -- you had to change this type here in order to keep things compiling. But there are other applications out there which use `kj::newHttpClient` and they may be doing things similar to this. So we cannot change the signature in a way that potentially breaks callers.

I have reverted the capnp PR since it may have broken people (indeed, the workerd build would have been broken as soon as someone tried to update the capnp submodule).

We'll need to go back and implement that the way I suggested originally: Factor out an abstract interface from `TlsContext` (not `TlsNetwork`), and add a new option to `HttpClientOptions`.

Sorry for the churn.

Reply:
Oh sorry, that's 100% my bad, you don't need to be sorry.

Reply:
That said, I am confused about what you mean. Can you elaborate on how the custom TlsContext should work and where I would use it?

Reply:
Okay, I think I get it.

* HttpClientSettings gets a new field: `kj::Maybe<TlsContext>`
* I can then use that in NetworkHttpClient to get a TlsNetwork via wrapNetwork
* Then I can use the TlsNetwork I just created to wrap the AsyncIoStream
* No API changes needed since the TlsContext lives in HttpClientSettings

But then I am confused why you are suggesting I need to factor out an abstract interface from TlsContext? Is it just so that we are explicit about which API calls can be made on that TlsContext?